docs/src/test/java/docs/http/javadsl/HttpsExamplesDocTest.java

```diff
@@ -19,14 +21,18 @@ public void testConstructRequest() {
     String unsafeHost = "example.com";
     //#disable-hostname-verification-connection
     final ActorSystem system = ActorSystem.create();
-    final ActorMaterializer mat = ActorMaterializer.create(system);
     final Http http = Http.get(system);
 
-    // WARNING: disabling hostname verification is a very bad idea, please don't unless you have a very good reason to.
-    final AkkaSSLConfig defaultSSLConfig = AkkaSSLConfig.get(system);
-    final AkkaSSLConfig badSslConfig = defaultSSLConfig
-      .convertSettings(s -> s.withLoose(s.loose().withDisableHostnameVerification(true)));
-    final HttpsConnectionContext badCtx = http.createClientHttpsContext(badSslConfig);
+    final HttpsConnectionContext badCtx = ConnectionContext.httpsClient((host, port) -> {
+      SSLEngine engine = SSLContext.getDefault().createSSLEngine(host, port);
+      engine.setUseClientMode(true);
+
+      // WARNING: this creates an SSL Engine without enabling endpoint identification/verification procedures
+      // Disabling host name verification is a very bad idea, please don't unless you have a very good reason to.
+      // When in doubt, use the `ConnectionContext.httpsClient` that takes an `SSLContext` instead.
```

Comment:
Maybe add why it disables hostname verification, i.e. that you otherwise need to call `setEndpoint...`.